packages/edit-post/src/components/preferences-modal/test/index.js

```diff
@@ -24,16 +24,20 @@ describe( 'EditPostPreferencesModal', () => {
 			useSelect.mockImplementation( () => [ true, true, false ] );
 			useViewportMatch.mockImplementation( () => true );
 			render( <EditPostPreferencesModal /> );
+			await screen.findByRole( 'tab', {
+				name: 'General',
+				selected: true,
+			} );
 			expect(
-				await screen.findByRole( 'dialog', { name: 'Preferences' } )
+				screen.getByRole( 'dialog', { name: 'Preferences' } )
 			).toMatchSnapshot();
 		} );
 		it( 'small viewports', async () => {
 			useSelect.mockImplementation( () => [ true, true, false ] );
 			useViewportMatch.mockImplementation( () => false );
 			render( <EditPostPreferencesModal /> );
 			expect(
-				await screen.findByRole( 'dialog', { name: 'Preferences' } )
```

Comment:
For small viewports, the modal uses the `Navigator*` components instead of `TabPanel`. The UI is completely synchronous, and the test doesn't need to `await`.